### PR TITLE
drop Addon.is_recommended property

### LINKED ISSUE #15368: drop Addon.is_recommended
it's unnecessary now we have Addon.promoted_groups (and Addon.promoted cached_property for ES)

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #15368

### CODE REVIEW

File: src/olympia/addons/models.py [hunk 1]
```diff
@@ -1292,21 +1292,12 @@ def _is_recommended_theme(self):
...
-    @cached_property
-    def is_recommended(self):
-        return (bool(self.promoted_group(group=RECOMMENDED)) or
-                self._is_recommended_theme())
-
-    def promoted_group(self, *, group=None, application=None,
-                       currently_approved=True):
-        """Is the addon currently promoted for the specified group and
-        application?
+    def promoted_group(self, *, application=None, currently_approved=True):
```
Comment (eviljeff) [inline src/olympia/addons/models.py:1295]: I also dropped `group` from this method - I realised there were two ways of doing the same thing, `if addon.promoted_group(group=FOO)` and `if addon.promoted_group() == FOO` which seemed redundant (and the group parameter was only used in tests anyway)

**Comment by muffinresearch** [comment]
Does the frontend need to been updated to account for this change first?

**Review by muffinresearch — APPROVED** [review]
r+ Looks sane to me, but I think it'll be good add this to @diox's post landing review queue, in-case I've missed something that needs more solid domain knowledge.

File: src/olympia/addons/serializers.py [hunk 2]
```diff
@@ -644,6 +653,11 @@ def fake_object(self, data):
             for data_author in data_authors
         ]
 
+        if 'is_recommended' in data:
+            # if we have this property in the index, store it so we can expose
+            # it in get_is_recommended.
+            obj._is_recommended = data['is_recommended']
```
Comment (diox) [inline src/olympia/addons/serializers.py:659]: Do we have a follow-up issue to remove this (and the code using it above) later ?

Comment (eviljeff) [reply src/olympia/addons/serializers.py:659]: https://github.com/mozilla/addons-server/issues/15271

**Pull request merged**